libraries/chain/hotstuff/finalizer.cpp

```diff
@@ -0,0 +1,302 @@
+         // finalizers not configured anymore.
+         for (const auto& [pub_key, fsi] : inactive_safety_info) {
+            fc::raw::pack(persist_file, pub_key);
+            fc::raw::pack(persist_file, fsi);
+         }
+         inactive_safety_info_written = true;
+      }
+      persist_file.flush();
+   } catch (const fc::exception& e) {
+      edump((e.to_detail_string()));
+      throw;
+   } catch (const std::exception& e) {
+      edump((e.what()));
+      throw;
+   }
+}
+
+// ----------------------------------------------------------------------------------------
+finalizer_set::fsi_map finalizer_set::load_finalizer_safety_info() {
+   fsi_map res;
+
+   EOS_ASSERT(!persist_file_path.empty(), finalizer_safety_exception,
+              "path for storing finalizer safety persistence file not specified");
+   EOS_ASSERT(!persist_file.is_open(), finalizer_safety_exception,
+              "Trying to read an already open finalizer safety persistence file: ${p}",
+              ("p", persist_file_path));
+
+   if (!std::filesystem::exists(persist_file_path)) {
+      elog( "unable to open finalizer safety persistence file ${p}, file doesn't exist",
+            ("p", persist_file_path));
+      return res;
+   }
+
+   persist_file.set_file_path(persist_file_path);
+
+   try {
+      // if we can't open the finalizer safety file, we return an empty map.
+      persist_file.open(fc::cfile::update_rw_mode);
+   } catch(std::exception& e) {
+      elog( "unable to open finalizer safety persistence file ${p}, using defaults. Exception: ${e}",
+            ("p", persist_file_path)("e", e.what()));
+      return res;
+   } catch(...) {
+      elog( "unable to open finalizer safety persistence file ${p}, using defaults", ("p", persist_file_path));
+      return res;
+   }
+   EOS_ASSERT(persist_file.is_open(), finalizer_safety_exception,
+              "unable to open finalizer safety persistence file: ${p}", ("p", persist_file_path));
+   try {
+      persist_file.seek(0);
+      uint64_t magic = 0;
+      fc::raw::unpack(persist_file, magic);
+      EOS_ASSERT(magic == fsi_t::magic, finalizer_safety_exception,
+                 "bad magic number in finalizer safety persistence file: ${p}", ("p", persist_file_path));
+      uint64_t num_finalizers {0};
+      fc::raw::unpack(persist_file, num_finalizers);
+      for (size_t i=0; i<num_finalizers; ++i) {
+         fsi_map::value_type entry;
+         fc::raw::unpack(persist_file, const_cast<bls_public_key&>(entry.first));
+         fc::raw::unpack(persist_file, entry.second);
+         res.insert(entry);
+      }
+      persist_file.close();
+   } catch (const fc::exception& e) {
+      edump((e.to_detail_string()));
+      // std::filesystem::remove(persist_file_path); // don't remove file we can't load
+      throw;
+   } catch (const std::exception& e) {
+      edump((e.what()));
+      // std::filesystem::remove(persist_file_path); // don't rremove file we can't load
```

Comment:
Remove the commented-out code but keep the comment.

Reply:
done